py/farm_ng/core/events_file_reader.py

```diff
@@ -135,49 +157,46 @@ def _build_event_index(self) -> None:
         self._reset_event_index()
         while True:
             try:
-                event = self._read_next_event()
-                current_offset = file_stream.tell()
-                self._event_index[uri_to_string(event.uri)].append(
-                    (event, current_offset)
-                )
-                self._skip_next_message(event)
+                event_log: EventLogPosition = self.read_next_event()
+                self._event_index.append(event_log)
+                self._skip_next_message(event_log.event)
             except EOFError:
                 break
 
         file_stream.seek(0)
 
-    def events(self, uri: Uri) -> List[Tuple[Event, int]]:
-        if not self._event_index:
+    def get_events(self, uri: Uri) -> List[EventLogPosition]:
+        if len(self._event_index) == 0:
             self._build_event_index()
-        return self._event_index[uri_to_string(uri)]
+        return [x for x in self.event_index if x.event.uri == uri]
 
     def num_events(self, uri: Uri) -> int:
         if not self.has_uri(uri):
             return 0
-        return len(self.events(uri))
+        return len(self.get_events(uri))
 
-    def get_event(self, uri: Uri, frame_id: int) -> Tuple[Event, int]:
-        assert self.has_uri(uri)
-        assert frame_id < self.num_events(uri)
-        return self.events(uri)[frame_id]
+    def get_index(self) -> List[EventLogPosition]:
+        if len(self._event_index) == 0:
+            self._build_event_index()
+        return self._event_index
 
-    def read_message(self, event: Event, offset: Optional[int] = None) -> Message:
+    def read_message(self, event_log: EventLogPosition) -> Message:
         file_stream = cast(IO, self._file_stream)
-        if offset is not None:
-            file_stream.seek(offset, 0)
-        name, package = parse_protobuf_descriptor(event.uri)
+        file_stream.seek(event_log.pos, 0)
+
+        name, package = parse_protobuf_descriptor(event_log.event.uri)
         message_cls = getattr(importlib.import_module(package), name)
 
-        payload: bytes = file_stream.read(event.payload_length)
+        payload: bytes = file_stream.read(event_log.event.payload_length)
 
         message: Message = message_cls()
         message.ParseFromString(payload)
         return message
 
     def read(self) -> Tuple[Event, Message]:
         assert self.is_open()
-        event = self._read_next_event()
-        return event, self.read_message(event)
+        event_log: EventLogPosition = self.read_next_event()
+        return event_log.event, self.read_message(event_log)
```

Comment:
`event_log` could be confusing for a single event
```suggestion
        event: EventLogPosition = self.read_next_event()
        return event.event, self.read_message(event)
```